pystiche_papers/sanakoyeu_et_al_2018/_modules.py

```diff
@@ -184,3 +192,202 @@ def residual_block(channels: int) -> ResidualBlock:
     )
 
     return ResidualBlock(residual)
+
+
+def get_transformation_block(
+    in_channels: int,
+    kernel_size: Union[Tuple[int, int], int],
+    stride: Union[Tuple[int, int], int],
+    padding: Union[Tuple[int, int], int],
+    impl_params: bool = True,
+) -> nn.Module:
+    if impl_params:
+        return nn.AvgPool2d(kernel_size=kernel_size, stride=stride, padding=padding)
+    return nn.Conv2d(in_channels, 3, kernel_size, stride=stride, padding=padding)
+
+
+class TransformerBlock(nn.Module):
+    r"""TransformerBlock from :cite:`SKL+2018`.
+
+    This block takes an image as input and produce a transformed image of the same size.
+
+    Args:
+        in_channels: Number of channels in the input. Defaults to ``3``.
+        kernel_size: Size of the convolving kernel. Defaults to ``10``.
+        stride: Stride of the convolution. Defaults to ``1``.
+        padding: Padding of the input. It can be either ``"valid"`` for no padding or
+            ``"same"`` for padding to preserve the size. Defaults to ``"same"``.
+        impl_params: If ``True``, use the parameters used in the reference
+            implementation of the original authors rather than what is described in
+            the paper.
+
+    If ``impl_params is True``, an :class:`~torch.nn.AvgPool2d` is used instead of a
+    :class:`~torch.nn.Conv2d` with :func:`~torch.nn.utils.weight_norm`.
+    """
+
+    def __init__(
+        self,
+        in_channels: int = 3,
+        kernel_size: Union[Tuple[int, int], int] = 10,
+        stride: Union[Tuple[int, int], int] = 1,
+        padding: str = "same",
+        impl_params: bool = True,
+    ):
+        super().__init__()
+        self.impl_params = impl_params
+
+        padding = get_padding(padding, kernel_size)
+
+        self.forwardBlock = get_transformation_block(
+            in_channels, kernel_size, stride, padding, impl_params=impl_params,
+        )
+
+    def forward(self, input: torch.Tensor) -> torch.Tensor:
+        if self.impl_params:
+            return cast(torch.Tensor, self.forwardBlock(input))
+        else:
+            return cast(torch.Tensor, nn.utils.weight_norm(self.forwardBlock(input)))
+
+
+def discriminator_encoder_modules(
+    in_channels: int = 3, inplace: bool = True,
+) -> Dict[str, nn.Sequential]:
+    # FIXME:  if/when the Python interpreter will learn to accept the correct signature
+    # https://stackoverflow.com/questions/41207128/how-do-i-specify-ordereddict-k-v-types-for-mypy-type-annotation
```

Comment:
Yeah, this is only possible for Python `>=3.7`. Lets ignore all errors for now and simply remove the ignores as soon as we bump the minimal required python version.

```suggestion
    # FIXME:  if/when the Python interpreter will learn to accept the correct signature
    # https://stackoverflow.com/questions/41207128/how-do-i-specify-ordereddict-k-v-types-for-mypy-type-annotation
```